Improvements for secure registration and login

jo-sf opened the pull request:
This is a revised version of PR #8619 based upon the current Joomla staging version.

Assume the following setup:
- the Joomla server is reachable via HTTP and HTTPS
- the Joomla administration is accessed via HTTPS ("Force HTTPS" is set to "Administrator Only")
- the Joomla site shall be accessed by default via HTTP
- optionally the ports used for HTTP and HTTPS are non-standard and shall be configurable

This PR aims at the following:
- there should be a single place for activating secure login and registration
- user registration data shall be transferred via HTTPS to the server if activated
- user login at the frontend shall always use HTTPS if activated
- all pages shown when logged in at the frontend shall always be transferred via HTTPS if activated
- user logout at the frontend shall always return to HTTP if secure login is activated
- site preview from the administration shall always open the site home page in HTTP mode
- switching between HTTP and HTTPS shall work even when non-standard ports are used
- if "Force HTTPS" is set to None no switching between HTTP and HTTPS shall be done

Currently secure login can be activated in each `mod_login` instance separately (and there are at least three of them in the demo site delivered with Joomla), and there is no way to enable secure login when using the login form menu item coming from `com_users`. Moreover it is not possible to activate secure transfer for user registration data which include sensitive data like password and mail address. Also the reminding and resetting functions shall be able to use HTTPS for transferring e.g. the mail address (currently HTTP is used for that).

In order to achieve this I moved the "usesecure" setting from the `mod_login` instance settings to the `com_users` configuration, and I changed all affected files (including message files) accordingly. In order to make foreign templates that override the `mod_login` login form work as before (assuming they check for the "usesecure" setting when logging in) I copy the "usesecure" setting from the `com_users` configuration dynamically to the `mod_login` instance settings prior to activating the login form of the template.

For proper links from the backend running in HTTPS mode to the site (site preview) I modified the administrator templates hathor and isis accordingly, moreover I had to change the administrator module `mod_status` for that purpose.

In order to handle internal URL checking even in the case of switching between HTTP and HTTPS I had to change the `JUri::isInternal()` method.

I order to make switching between HTTP and HTTPS protocols easier (same lines of code needed at many places) I implemented two new methods `JUri::siteScheme()` and `JUri::adminScheme()` which do this switching.

If you enable "Force HTTPS" by setting it to either "Administrator Only" or "Entire Site" (Global Configuration, Server Settings) you'll now get two additional fields for the HTTP and the HTTPS ports where you can enter the ports used at your site. This is necessary only when using non-standard ports like 8080 for HTTP and 8443 for HTTPS.

If you change these ports they are checked as well as the "Force HTTPS" mode itself, and in case of an error doing so the change is undone.

At all code locations where switching between HTTP and HTTPS is performed these ports if set are used accordingly.

Note that switching between HTTP and HTTPS protocols is done only if "Force HTTPS" is set to anything else than "None". This should help in cases like in PR #11213.

The login and logout forms in `mod_login` used the current URL as their target for the forms which is of no great use since the real target of these forms is passed in the hidden field "return" I changed that back to "index.php" as it was about two years ago.

I saw that the beez3 template contains an override for the `mod_login` login form but no separate override for the logout form. Instead the login form still contained the login and logout cases as was the case in `mod_login` until more than three years ago (August 2012).

Finally I added the new field attribute "renderon" (see PR #11207) both to the "usesecure" field in `com_users` and to the "secure" field in `com_menus`, these fields are shown only if the global flag "force_ssl" is set to 1 ("Administrator Only") and if PR #11207 has been applied. My reason for this is that if "force_ssl" is 0 probably no HTTPS is available such that switching to HTTPS would most likely fail, and if "force_ssl" is 2 ("Entire Site") all pages are already shown via HTTPS such that no switching is necessary.
#### Testing Instructions
-  set up current Joomla version including demo site and apply all changes from this PR
- optionally apply the changes from PR #11207 
- the webserver must be reachable via HTTP and HTTPS
- log in to the backend as administrator
- set "Force HTTPS" in Global Configuration, Server Settings to "Administrator Only"
- the backend should now be accessed via HTTPS only
- all site preview links should point to HTTP (hathor or isis template assumed)
- frontend login (regardless which) continue in HTTP mode
- go to the Users Configuration and enable "Encrypt Login / Registration Forms"
- check registration form: the action URL should point to HTTPS
- after successful registration the user will be back in HTTP mode
- check frontend login form (regardless which): the action URL should point to HTTPS
- after successful login all pages should be displayed via HTTPS
- in case of a login failure the user should be back in HTTP mode
- check frontend logout form: the action URL should point to HTTPS
- perform logout: the user should be back in HTTP mode
- check remind and reset links: the action URL for both of them should point to HTTPS
- after performing remind / reset functions the user should be back in HTTP mode
- set any menu item to "Secure" mode (tab "Metadata")
- check in the frontend that this menu item now points to HTTPS
- configure additional server ports usable for HTTP and HTTPS, e.g. to 8080 and 8443
- configure these ports in Global Configuration, Server Settings (below "Force HTTPS")
- check that switching between HTTP and HTTPS works as expected
- check that the action URLs mentioned above contain the right HTTPS port
- set "Force HTTPS" in Global Configuration, Server Settings to "Entire Site"
- all pages should now be served via HTTPS, no switching when logging in or out
- if PR #11207 was applied too:
  - the field "Encrypt Login / Registration Forms" in the Users Configuration is no longer rendered
  - the field "Secure" in menu items (tab "Metadata") is no longer rendered

Note: In testing this PR you might encounter that your browser uses the HTTPS protocol even if you changed "Force HTTPS" back from "Entire Site" to "Administrator Only" or "None" or back from "Administrator Only" to "None". This is not a problem with these changes but is caused by the browser caching redirections it received prior to changing the "Force HTTPS" setting. In such cases please close all windows or tabs showing content from the Joomla instance used for testing, clear the browser's cache and then try again.

jo-sf committed:
Messages for HTTP/HTTPS port fields

jo-sf committed:
Add messages regarding encrypted login

jo-sf committed:
Remove messages regarding encrypted login

jo-sf committed:
Fields for HTTP/HTTPS ports

jo-sf committed:
New field "Encrypt Login / Registration Forms"

jo-sf committed:
Field "Encrypt Login Form" removed

jo-sf committed:
Remove messages regarding encrypted login

jo-sf committed:
Field "Encrypt Login Form" removed

jo-sf committed:
Changes due to HTTP/HTTPS ports

Changes in base() and isInternal(): use appropriate port if switching to another scheme; new methods siteScheme() and adminScheme(), both methods set the appropriate scheme and port if not set already

jo-sf committed:
Take flag "usesecure" from "com_users" component

jo-sf committed:
Take flag "usesecure" from "com_users" component

jo-sf committed:
Change preview URI construction

jo-sf committed:
Change preview URI construction

jo-sf committed:
Change preview URI construction

jo-sf committed:
Change preview URI construction

jo-sf committed:
Change preview URI construction

jo-sf committed:
Check "usesecure" flag

If the "usesecure" flag is set the registration was done via HTTPS, afterwards route back to HTTP protocol

jo-sf committed:
Check "usesecure" flag

If the "usesecure" flag is set the reminder function was done via HTTPS, afterwards route back to HTTP protocol

jo-sf committed:
Check "usesecure" flag

If the "usesecure" flag is set the password reset function was done via HTTPS, afterwards route back to HTTP protocol

jo-sf committed:
Check "usesecure" flag

If the "usesecure" flag is set login data was sent via HTTPS, in case of a failed login route back to HTTP protocol

jo-sf committed:
Handle URI objects too

The methods siteScheme() and adminScheme() now handle URI strings and URI objects and return URI strings or URI objects accordingly

jo-sf committed:
Use JUri::siteScheme() for links in mails

Build appropriate links for mails sent to the user and the administrator, use JUri::siteScheme() for setting up appropriate scheme and port

jo-sf committed:
Check "usesecure" flag

Switch to HTTPS when logging in if "usesecure" is set

jo-sf committed:
Check "usesecure" flag

Log out via HTTPS

jo-sf committed:
Check "usesecure" flag

Register via HTTPS

jo-sf committed:
Check "usesecure" flag

Reminder function via HTTPS

jo-sf committed:
Check "usesecure" flag

Password reset via HTTPS

jo-sf committed:
Check "usesecure" flag

Password reset via HTTPS

jo-sf committed:
Check "usesecure" flag

Password reset via HTTPS

jo-sf committed:
Field "usesecure": attribute "renderon" added

Attribute "renderon" added to field "usesecure" (see PR #11207) - field shall be shown only if "Force HTTPS" is set to "Administrator Only"

jo-sf committed:
Forward to HTTPS: set port too

jo-sf committed:
Forward to HTTPS: set port too

jo-sf committed:
Secure/unsecure URL building changed

Getting the secure/unsecure URLs now uses JUri::siteScheme().

jo-sf committed:
Use appropriate scheme for return URL

jo-sf committed:
Copy "usesecure" from com_users to mod_login params

jo-sf committed:
Login form action set to "index.php"

Login form action set to "index.php" since it is not really relevant which URL is used here since the real target is encoded in the hidden field "return", moreover JRoute::_() is only able to handle URLs that start with "index.php"; the "usesecure" flag now comes from the com_users component.

jo-sf committed:
Logout form action set to "index.php"

Logout form action set to "index.php" since it is not really relevant which URL is used here since the real target is encoded in the hidden field "return", moreover JRoute::_() is only able to handle URLs that start with "index.php"; the "usesecure" flag now comes from the com_users component.

jo-sf committed:
Separate "logout" from "login"

jo-sf committed:
Separate "logout" from "login"

jo-sf committed:
Field "secure": attribute "renderon" added

Attribute "renderon" added to field "secure" (see PR #11207) - field shall be shown only if "Force HTTPS" is set to "Administrator Only"

jo-sf committed:
Switch to HTTP if $ssl is 2

jo-sf committed:
Translate "secure" flag into $ssl for JRoute::_()

jo-sf committed:
Messages for Force HTTPS and HTTP/HTTPS port checks

jo-sf committed:
More checks needed (Force HTTPS, HTTP(S) ports)

If "Force HTTPS" is set we need to check that the server is available via HTTPS with the HTTPS port probably defined, reset values that might have caused an error. If a HTTP port is set we need to check that the server is available via HTTP with this port, reset value in case of an error. Note that both the HTTP and the HTTPS port are used only if "Force HTTPS" is not set to "None" (0).

zero-24 commented on administrator/components/com_config/model/application.php:
```diff
@@ -118,33 +119,91 @@ public function save($data)
+		// Check if we can set the HTTP port (do this check only if Force HTTPS is set)
+		if ((int) $data['force_ssl'] !== 0 && (int) $data['http_port'] !== 0 && (int) $data['http_port'] !== (int) $config->get('http_port', '0'))
+		{
+			try:
```
do you expoect to use `try:` an not `try`

jo-sf replied:
yes - typo

jo-sf committed:
Typo and code style errors

jo-sf committed:
Typo (dot instead of comma)

jo-sf committed:
Typo (dot instead of comma)

jo-sf committed:
Code style

andrepereiradasilva commented on administrator/templates/hathor/login.php:
```diff
@@ -14,8 +14,7 @@
 // Gets the FrontEnd Main page Uri
-$frontEndUri = JUri::getInstance(JUri::root());
-$frontEndUri->setScheme(((int) JFactory::getApplication()->get('force_ssl', 0) === 2) ? 'https' : 'http');
+$siteUri = JUri::siteScheme(JUri::root());
```
i made a PR so you could use JRoute for this
See https://github.com/joomla/joomla-cms/pull/11060

andrepereiradasilva commented on administrator/modules/mod_login/mod_login.xml:
```diff
@@ -21,18 +21,6 @@
-					<option value="1">JYES</option>
-					<option value="0">JNO</option>
-				</field>
-			</fieldset>
```
is removing a parameter B/C ?

jo-sf replied:
Well it isn't B/C, but I decided to to remove this parameter because

* you had to set this parameter individually for every login module (and e.g. with the samle data there are three such login modules)
* there is no way to set "usesecure" for the login form defined in com_users (this form is shown automatically if you enter wrong data in the login form defined in the login module)
* there should be a single place where you can enable/disable secure login for all login forms

andrepereiradasilva replied:
i understand the point. but if is not done in a B/C way i think it can only go for 4.0

jo-sf replied:
Probably I can modify this such that the "usesecure" parameter here and in modules/mod_login/mod_login.xml can stay (up to version 4.0 or so) but with lesser priority than any "Force HTTPS" setting and the "usesecure" setting in com_users.

But this will need some days for checking and testing... hope that's OK

andrepereiradasilva replied:
You could add an option use global thst uses com users value

andrepereiradasilva replied:
maybe
```xml
<field
	name="usesecure"
 	type="radio"
 	label="MOD_LOGIN_FIELD_USESECURE_LABEL"
 	description="MOD_LOGIN_FIELD_USESECURE_DESC"
 	class="btn-group btn-group-yesno"
 	default="0"
 	>
 	<option value="">JGLOBAL_USE_GLOBAL</option>
 	<option value="1">JYES</option>
 	<option value="0">JNO</option>
</field>
```

And them load it
```php
$params->get('usesecure', JComponentHelper::getParams('com_users')->get('usesecure', 0));
```

andrepereiradasilva commented on administrator/components/com_config/model/form/application.xml:
```diff
@@ -741,6 +741,24 @@
+			description="COM_CONFIG_FIELD_HTTP_PORT_DESC"
+			filter="string"
+			showon="force_ssl:1,2"
+			size="8" />
```
maybe
```xml
		<field
			name="http_port"
			type="integer"
			label="COM_CONFIG_FIELD_HTTP_PORT_LABEL"
			description="COM_CONFIG_FIELD_HTTP_PORT_DESC"
			first="1"
			last="65536"
			step="1"
			default="80"
			hint="80"
			validate="number"
			showon="force_ssl:0,1"
			size="8"
		/>
```
same for https

jo-sf replied:
At the first sight it looks like a good idea but the result in the browser is sort of a problem: you get a selection box with all numbers between 1 and 65535 (maximum port number allowed), and loading the global configuration page is very slow even when your server is in the local network.

Moreover all other fields where port numbers are asked for (like FTP port, SMTP port) are defined the same way as these two new fields, which means: text fields without any check. If there is any other form field type that allows to

* show a text input field
* checks the typed in data for being a valid number
 * if possible in the browser
 * in every case prior to storing it in the configuration

then the definition of all such form fields should be changed accordingly.

andrepereiradasilva replied:
sorry i mean
```xml
<field
	name="http_port"
	type="number"
	label="COM_CONFIG_FIELD_HTTP_PORT_LABEL"
	description="COM_CONFIG_FIELD_HTTP_PORT_DESC"
	min="1"
	max="65536"
	default="80"
	hint="80"
	filter="integer"
	validate="number"
	showon="force_ssl:1,2"
	size="6"
/>
```
just like the session tim ein global config

conconnl commented:
@jo-sf 
There is an error in the patch.
The patch changes the things needed but the registration button stops working.
Because he tries to do a post to: https:///index.php...............

Have not tested the rest, but this will probably happen too on the login button.

<img width="933" alt="screen shot 2016-07-26 at 23 53 45" src="https://cloud.githubusercontent.com/assets/11913918/17156954/c1c0777a-538c-11e6-8a9f-8e92692b4bdf.png">

jo-sf committed:
Check $force_ssl and set host

Scheme and port may only be set if "Force HTTPS" is not "None" ($force_ssl != 0), in this case we must set the host prior to calling JUri::siteScheme().

jo-sf committed:
Set scheme and port always

Do not check whether the desired scheme is set already, set scheme and port always (might not yet be set at all)

jo-sf commented:
@conconnl 
You're absolutely right. I optimized too much in JRoute::_() which caused the host name to be missing in the resulting URL - it should now work as expected.

Please note that switching to HTTPS when sending the registration form to the server and when logging in is done only when "Force HTTPS" is set to "Administrator Only"; if it is set to "None" no switching is done, and if it is set to "Entire Site" no switching is necessary since the registration form has already been loaded via HTTPS.

Unfortunately this PR is no longer automatically mergeable with the staging code and it seems to cause some conflicts when testing it by means of PatchTester (at least some changes done in `administrator/components/com_config/model/application.php` coming from PR #11189 are lost after applying this PR to the staging code).

Can I and if yes how can I cure this?

brianteeman commented:
You will need to update your own branch to the latest staging and if necessary resolve any merge conflicts there.

andrepereiradasilva commented on administrator/templates/hathor/html/mod_login/default.php:
```diff
@@ -10,8 +10,9 @@
+$usersConfig = JComponentHelper::getParams('com_users');
 ?>
-<form action="<?php echo JRoute::_('index.php', true, $params->get('usesecure')); ?>" method="post" id="form-login">
+<form action="<?php echo JRoute::_('index.php', true, $usersConfig->get('usesecure')); ?>" method="post" id="form-login">
```
save as above ? 
```php
$params->get('usesecure', JComponentHelper::getParams('com_users')->get('usesecure', 0));
```

jo-sf commented:
@brianteeman 
Sorry for being ignorant, but how can I update my own branch (I think you mean the "patch-XX" branch from where I created this PR) to the latest staging (here I assume you mean the latest "joomla/joomla-cms" staging)? I searched for any way to do this in the github web interface, but I didn't find any way to do so. Is this at all possible via the web interface?

brianteeman commented:
Yeah you guessed what I meant correctly and no its not possible
autoamtically via the web interface as far as I know.

I am going out for a few hours but will take a deeper look at this for you
when I get back (unless someone else helps in the meantime)

On 29 July 2016 at 07:57, jo-sf <notifications@github.com> wrote:

> @brianteeman <https://github.com/brianteeman>
> Sorry for being ignorant, but how can I update my own branch (I think you
> mean the "patch-XX" branch from where I created this PR) to the latest
> staging (here I assume you mean the latest "joomla/joomla-cms" staging)? I
> searched for any way to do this in the github web interface, but I didn't
> find any way to do so. Is this at all possible via the web interface?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/11304#issuecomment-236108707>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABPH8aP5mXsLeL_xAx8onfmFlrcKsbiSks5qaaQ_gaJpZM4JUeEY>
> .
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

jo-sf commented:
@conconnl 
Prior to continuing with this PR (branch update, bug fixing, improvements) I think we have to make some decisions. @andrepereiradasilva made a point asking for B/C - this is something I had only partly in mind when creating this PR.

Let's first look at all places where you can switch to secure mode in Joomla today (those are at least all places I found so far):

* you can set "Force HTTPS" in the global configuration to "Administrator Only" or "Entire Site"
 * since Joomla 3.6.0 this is checked whether you can access your server via HTTPS, and if this is not the case "Force HTTPS" is reset to None
* you can set "Encrypt Login Form" for each site instance of the `mod_login` module individually to "Yes"
* you can set "Encrypt Login Form" for the administrator instance of the `mod_login` module to "Yes"
 * note that with this setting you can effectively lock you out of your Joomla administration access, and the only way to undo this is by making a change in the database
* you can set "Secure" in the "Metadata" tab of most menu items to "On" or "Off"

My first intention when implementing this PR was to eliminate the "Encrypt Login Form" settings in the `mod_login` modules both for the site and the administrator and to replace the site setting by means of a new "Encrypt Login / Registration Forms" setting in `com_users`. The "Secure" setting in the menu items should only be available when "Force HTTPS" is set to "Administrator Only".

But now triggered by some of @andrepereiradasilva's remarks I think we have to split this into two parts, one that is B/C and continues to contain and honor all existing settings and one that will come with the next major Joomla release (Joomla 4) and that will remove some of the then obsolete or problematic settings. 

The Joomla 3.6/3.7 implementation of this PR should IMO contain the following changes:

* the new "Encrypt Login / Registration Forms" setting in `com_users` superseding the still existing "Encrypt Login Form" settings in the site instances of the `mod_login` module
 * if this new setting is "Yes" site login and registration forms will be encrypted (this means that the contents of these forms will be sent via HTTPS to the server)
 * if this new setting is "No" login forms will be encrypted if the settings for these forms (`mod_login` modules) say so
 * this new "Encrypt Login / Registration Forms" setting will be shown only when "Force HTTPS" is set to "Administrator Only"
  * if "Force HTTPS" is set to "None" the existence of HTTPS is not guaranteed and tested
  * if "Force HTTPS" is set to "Entire Site" all pages are shown via HTTPS
* the "Encrypt Login Form" settings in site and administrator `mod_login` modules continue to exist
 * the setting in site `mod_login` modules will be shown only when "Force HTTPS" is not "Entire Site"
 * the setting in the administrator `mod_login` module will be shown only when "Force HTTPS" is "None"
* the "Secure" setting in the menu items continue to exist, it will be shown only when "Force HTTPS" is not "Entire Site"

The implementation for the next major release should contain the following changes with respect to the changes listes above for Joomla 3.6/3.7:

* the "Encrypt Login Form" settings in site and administrator `mod_login` modules will be removed
 * for the site `mod_login` modules the "Encrypt Login / Registration Forms" setting from `com_users` will be dynamically copied to the module's settings such that templates overriding the form's HTML code continue to work as expected
 * the administrator `mod_login` module no longer needs to switch to HTTPS, instead you have to set "Force HTTPS" to "Administrator Only" (or "Entire Site")
* the "Secure" setting in the menu items will be shown only when "Force HTTPS" is "Administrator Only"

What do you (and others) think about that?

conconnl commented:
@jo-sf 
It's very important to not have a B/C break when making PRs for current major versions.
So splitting the PR is the way to go, as you describe yourself first adding features to the current branch and afterwards creating a PR for the next major (4.x or other number) to remove deprecated settings.  

I think this does not mean you can hide the setting in mod_login when you have the Secure Form setting enabled in the User Configuration.
This only means you cannot remove the setting to keep the sites who upgrade working and make people can still find the setting to secure the form when choosing to only secure the administrator.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11304">issues.joomla.org/joomla-cms/11304</a>.</sub>

andrepereiradasilva commented:
@jo-sf i would also remove the link replacing in the templates files from this PR
because i think this is not the best way to do it. I proposed a patch to add the flexibility to JRoute (https://github.com/joomla/joomla-cms/pull/11060) so we could do that whenever needed.

conconnl commented:
@jo-sf 
The "perform logout: the user should be back in HTTP mode" step does not go back to HTTP, the site stays in HTTPS<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11304">issues.joomla.org/joomla-cms/11304</a>.</sub>

andrepereiradasilva commented:
BTW @jo-sf you may also want to check this PRs i made https://github.com/joomla/joomla-cms/pull/10630 and https://github.com/joomla/joomla-cms/pull/10466

jo-sf committed:
merge with upstream/staging

jo-sf commented:
@conconnl @andrepereiradasilva 
Based mainly upon your comments I think I need to redesign this PR - it's too large to be tested and reviewed thoroughly, it's very "colored" (9 tags attached in issues.joomla.org) and it's not B/C.

Unfortunately this means that you'll have to wait some more time until the resulting PRs can be tested and reviewed. I'm not yet sure how many this will be, but I expect three or so - and at least another one for the next major release that contains all the non B/C stuff.

brianteeman commented:
Shall we close this here for now then?

andrepereiradasilva commented:
@jo-sf no problem take your time. There is not like there is no other work to be done were :)

jo-sf commented:
@brianteeman if this is not a problem then please let this PR open until I managed to create the new PRs. This way I can use this PR and all the comments from you all as my source of inspiration...

brianteeman commented:
It will still be accessible to you as a closed issue. But by closing it we
prevent people wasting their time commening or testing it

On 2 August 2016 at 20:06, jo-sf <notifications@github.com> wrote:

> @brianteeman <https://github.com/brianteeman> if this is not a problem
> then please let this PR open until I managed to create the new PRs. This
> way I can use this PR and all the comments from you all as my source of
> inspiration...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/11304#issuecomment-237009817>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABPH8d7qBPOkVvyUqofhlDWaUTsMsEZ_ks5qb5U7gaJpZM4JUeEY>
> .
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

jo-sf commented:
@brianteeman 
Yes it would really be wasting of time if someone would continue to test this PR since it will never be merged into the base branch. Moreover my PR is no longer working as I expected it to do (see e.g. the comment from @conconnl some days ago), and I see no good reason to inspect this further.

But commenting is (I think) another point. Maybe someone seeing this PR would like to comment on it although I plan to redesign it, moreover I thought of adding links to the redesigned PRs to this PR as soon as they are ready to be tested.

What is your standard procedure in such cases?

brianteeman commented:
Usually we would close this. If/when you add links to the new PR to this
issue everyone subscribed to this issue would be notified of the new links

On 3 August 2016 at 08:14, jo-sf <notifications@github.com> wrote:

> @brianteeman <https://github.com/brianteeman>
> Yes it would really be wasting of time if someone would continue to test
> this PR since it will never be merged into the base branch. Moreover my PR
> is no longer working as I expected it to do (see e.g. the comment from
> @conconnl <https://github.com/conconnl> some days ago), and I see no good
> reason to inspect this further.
>
> But commenting is (I think) another point. Maybe someone seeing this PR
> would like to comment on it although I plan to redesign it, moreover I
> thought of adding links to the redesigned PRs to this PR as soon as they
> are ready to be tested.
>
> What is your standard procedure in such cases?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/11304#issuecomment-237161126>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABPH8e-El5AYd177RDPwf67uYMrsg8c_ks5qcD_ugaJpZM4JUeEY>
> .
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

jo-sf commented:
@brianteeman 
I wasn't sure whether I can add additional comments to a closed issue, but if that works and everybody subscribed to this issue will receive a notification you might safely close this issue now.

BTW may I also add a link to a closed issue to a new issue?

brianteeman commented:
Closed. You can still comment if needed
